paddle/operators/multiplex_op.h

```diff
@@ -0,0 +1,72 @@
+
+/* Copyright (c) 2016 PaddlePaddle Authors. All Rights Reserve.
+
+   Licensed under the Apache License, Version 2.0 (the "License");
+   you may not use this file except in compliance with the License.
+   You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+   Unless required by applicable law or agreed to in writing, software
+   distributed under the License is distributed on an "AS IS" BASIS,
+   WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+   See the License for the specific language governing permissions and
+   limitations under the License. */
+
+#pragma once
+
+#include "paddle/framework/eigen.h"
+#include "paddle/framework/op_registry.h"
+
+namespace paddle {
+namespace operators {
+
+template <typename T>
+class MultiplexCPUKernel : public framework::OpKernel {
+ public:
+  void Compute(const framework::ExecutionContext& ctx) const {
+    auto ins = ctx.MultiInput<framework::Tensor>("X");
+    auto* out = ctx.Output<framework::LoDTensor>("Out");
+    out->mutable_data<T>(ctx.GetPlace());
+
+    auto index = ins[0]->data<T>();
+    auto rows = ins[1]->dims()[0];
+    auto cols = ins[1]->dims()[1];
+    for (auto i = 0; i < rows; i++) {
+      int k = (int)index[i] + 1;
+      memcpy(out->data<T>() + i * cols, ins[k]->data<T>() + i * cols,
```

Comment:
Maybe we can combine cpu code and cuda code in one file.
```
template <typename Place, typename T>
class MultiplexKernel : public framework::OpKernel 
```
We can use 
```
t.device(context.GetEigenDevice<Place>()) = t.constant(static_cast<T>(0));
```
for set cpu/gpu to zero

And we can use 
```
memory::Copy
```
 for both cpu/gpu copy

Reply:
Done

Reply:
It seems that merge CPU/GPU code together is not a good idea here. I make a mistake.
If CPU and GPU both use Eigen, we can reuse codes easily. But if not, it's actually better to split CPU and GPU implementation.

Reply:
Done. split CPU/GPU code again.